src/ethereum/networkAdapters/PulsechainScanAdapter.ts

```diff
@@ -0,0 +1,262 @@
+
+      this.currentScan = this.checkTransactions(params)
+      const update = await this.currentScan
+      return update
+    } catch (error) {
+      console.error(error)
+      throw error
+    } finally {
+      this.currentScan = undefined
+    }
+  }
+
+  private async checkTransactions(
+    params: GetTxsParams
+  ): Promise<EthereumNetworkUpdate> {
+    const { startBlock, currencyCode } = params
+    const address = this.ethEngine.walletLocalData.publicKey
+
+    const scanTransactions: PulsechainScanTransaction[] = []
+    let nextPageParams: NextPageParams | undefined
+
+    while (true) {
+      const responseData = await this.queryTransactions(address, nextPageParams)
+
+      scanTransactions.push(...responseData.items)
+      nextPageParams = responseData.next_page_params
+
+      // Exit if a transaction block height is less than/equal to the startBlock:
+      if (responseData.items.some(scanTx => scanTx.block <= startBlock)) break
+
+      // Exit if there is no next page of transactions:
+      if (nextPageParams == null) break
+    }
+
+    // Convert the transaction data into EdgeTransactions:
+
+    const edgeTransactions: EdgeTransaction[] = scanTransactions.map(tx =>
+      this.processScanTransaction(tx, currencyCode)
+    )
+
+    return {
+      tokenTxs: {
+        [currencyCode]: {
+          blockHeight: startBlock,
+          edgeTransactions
+        }
+      },
+      server: this.config.servers.join(',')
+    }
+  }
+
+  private processScanTransaction(
+    scanTx: PulsechainScanTransaction,
+    currencyCode: string
+  ): EdgeTransaction {
+    const ourReceiveAddresses: string[] = []
+
+    const txid = scanTx.hash
+    if (txid == null) {
+      throw new Error('Invalid transaction result format')
+    }
+
+    const isSpend =
+      scanTx.from.hash.toLowerCase() ===
+      this.ethEngine.walletLocalData.publicKey.toLowerCase()
+    const tokenTx = currencyCode !== this.ethEngine.currencyInfo.currencyCode
+
+    const gasPrice = scanTx.gas_price
+    const nativeNetworkFee: string =
+      gasPrice != null ? mul(gasPrice, scanTx.gas_used) : '0'
+
+    let nativeAmount: string
+    let networkFee: string
+    let parentNetworkFee: string | undefined
+
+    if (isSpend) {
+      if (tokenTx) {
+        nativeAmount = sub('0', scanTx.value)
+        networkFee = '0'
+        parentNetworkFee = nativeNetworkFee
+      } else {
+        // Spend to self. netNativeAmount is just the fee
+        if (scanTx.from.hash.toLowerCase() === scanTx.to.hash.toLowerCase()) {
+          nativeAmount = sub('0', nativeNetworkFee)
+          networkFee = nativeNetworkFee
+        } else {
+          nativeAmount = sub(sub('0', scanTx.value), nativeNetworkFee)
+          networkFee = nativeNetworkFee
+        }
+      }
+    } else {
+      // Receive
+      if (tokenTx) {
+        nativeAmount = scanTx.value
+        networkFee = '0'
+      } else {
+        nativeAmount = scanTx.value
+        networkFee = '0'
+      }
```

Comment:
Both cases look the same here.

Reply:
Fixed, and fixed the source of this copy-pasta